qiskit_optimization/applications/sk_model.py

```diff
@@ -0,0 +1,110 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An application class for the Sherrington Kirkpatrick (SK) model."""
+
+from typing import Union, List, Optional
+
+import networkx as nx
+import numpy as np
+from docplex.mp.model import Model
+
+from .graph_optimization_application import OptimizationApplication
+from ..algorithms import OptimizationResult
+from ..problems.quadratic_program import QuadraticProgram
+from ..translators import from_docplex_mp
+
+
+class SKModel(OptimizationApplication):
+    r"""Optimization application of the "Sherrington Kirkpatrick (SK) model" [1].
+
+    The SK Hamiltonian over n spins is given as:
+    :math:`H(x)=-1/\sqrt{n} \sum_{i<j} w_{i,j}x_ix_j`,
+    where :math:`x_i\in\{\pm 1\}` and :math:`w_{i,j}\in\{\pm 1\}` are chosen independently and uniformly
+    at random. Notice that there are other variants e.g., with :math:`w_{i,j}` chosen from the normal
+    distribution with mean 0 and variance 1.
+
+    References:
+        [1]: Dmitry Panchenko. "The Sherrington-Kirkpatrick model: an overview",
+        https://arxiv.org/abs/1211.1094
+    """
+
+    def __init__(
+        self, num_of_sites: int, rng_or_seed: Optional[Union[np.random.Generator, int]] = None
+    ):
+        """
+        Constructor for the SK model.
+
```

Comment:
```suggestion
```
When the html is built the class docstring and constructor docstring are merged.  We try to avoid having something like this, that states what the code tells you anyway.